src/task_list.dart

```diff
@@ -0,0 +1,92 @@
+import 'task.dart';
+import 'package:dolumns/dolumns.dart';
+import 'dart:io';
+import 'dart:core';
+
+TaskList taskList = TaskList();
+Task myTask = Task();
+
+class TaskList {
+  List<Task> taskList = [];
+  int id = 1;
+
+  //addTaskfunction
+  void addTask(Task myTask) {
+    myTask.id = id;
+    id++;
+
+    //promt user to enter title and dec
+    stdout.write("Enter Task title: ");
+    myTask.title = stdin.readLineSync();
+
+    stdout.write("Enter Task descrption: ");
+    myTask.description = stdin.readLineSync();
+
+    ///add tasks
+    taskList.add(myTask);
+  }
+
+  void isDone() {
+    //choice a task id
+    stdout.write("Enter Task ID to mark as complete: ");
+    int taskId = int.parse(stdin.readLineSync()!);
+    Task foundTask =
+        taskList.firstWhere((task) => task.id == taskId, orElse: () => Task());
+    // Check if the task was found
+    if (foundTask.id != -1) {
+      // Update the isDone property of the found task
+      foundTask.isDone = true;
+    } else {
+      print("not a valid id");
+    }
+  }
```

Comment:
should invert instead (to make it so i can mark it as undone)
```suggestion
  void isDone() {
    //choice a task id
    stdout.write("Enter Task ID to mark as complete: ");
    int taskId = int.parse(stdin.readLineSync()!);
    Task foundTask =
        taskList.firstWhere((task) => task.id == taskId, orElse: () => Task());
    // Check if the task was found
    if (foundTask.id != -1) {
      // Update the isDone property of the found task
      foundTask.isDone = !foundTask.isDone;
    } else {
      print("not a valid id");
    }
  }
```